src/code/sys_math3d.c

```diff
@@ -1368,93 +1620,312 @@ s32 Math3D_PointInCyl(Cylinder16* cyl, Vec3f* point) {
     static Sphere16 bottomSphere;
-    static Vec3f D_8016A740;
-    static Vec3f D_8016A750;
+    static Vec3f cylIntersectA;
+    static Vec3f cylIntersectB;
 
-    f32 sp8C;
+    f32 yIntersect;
     f32 cylTop;
     f32 cylBottom;
-    f32 phi_f2;
-    f32 t;
-    f32 temp_ret;
-    Vec3f sp6C;
-    Vec3f sp60;
-    Vec3f sp54;
-    f32 temp_f14_2;
-    f32 temp_f2;
+    f32 minDistSq;
+    f32 radiusTodistFromCylYIntersectTov0v1;
+    f32 distFromPointAToIntersectASq;
+    Vec3f cylIntersectCenter;
+    Vec3f midpointv0v1;
+    Vec3f diffMidpointIntersect;
+    f32 distFromCylYIntersectTov0v1;
+    s32 pad;
 
     cylBottom = (f32)cyl->pos.y + cyl->yShift;
     cylTop = cyl->height + cylBottom;
 
-    // If all of the verticies are below or all of the verticies are above the cylinder.
-
     if (((tri->vtx[0].y < cylBottom) && (tri->vtx[1].y < cylBottom) && (tri->vtx[2].y < cylBottom)) ||
         ((cylTop < tri->vtx[0].y) && (cylTop < tri->vtx[1].y) && (cylTop < tri->vtx[2].y))) {
-        return 0;
+        // If all of the verticies are below or all of the verticies are above the cylinder.
+        return false;
     }
-    phi_f2 = 1.e38f;
-    if (func_800CEE0C(cyl, &tri->vtx[0], &tri->vtx[1], &D_8016A740, &D_8016A750)) {
-        phi_f2 = func_800CB650(&D_8016A740, &tri->vtx[0]);
-        *intersect = D_8016A740;
+
+    minDistSq = 1.e38f;
+    if (Math3D_CylVsLineSeg(cyl, &tri->vtx[0], &tri->vtx[1], &cylIntersectA, &cylIntersectB)) {
+        distFromPointAToIntersectASq = Math3D_Vec3fDistSq(&cylIntersectA, &tri->vtx[0]);
+        minDistSq = distFromPointAToIntersectASq;
+        *intersect = cylIntersectA;
     }
 
-    if (func_800CEE0C(cyl, &tri->vtx[2], &tri->vtx[1], &D_8016A740, &D_8016A750)) {
-        temp_ret = func_800CB650(&D_8016A740, &tri->vtx[2]);
-        if (temp_ret < phi_f2) {
-            *intersect = D_8016A740;
-            phi_f2 = temp_ret;
+    if (Math3D_CylVsLineSeg(cyl, &tri->vtx[2], &tri->vtx[1], &cylIntersectA, &cylIntersectB)) {
+        distFromPointAToIntersectASq = Math3D_Vec3fDistSq(&cylIntersectA, &tri->vtx[2]);
+        if (distFromPointAToIntersectASq < minDistSq) {
+            *intersect = cylIntersectA;
+            minDistSq = distFromPointAToIntersectASq;
         }
     }
 
-    if (func_800CEE0C(cyl, &tri->vtx[0], &tri->vtx[2], &D_8016A740, &D_8016A750)) {
-        temp_ret = func_800CB650(&D_8016A740, &tri->vtx[0]);
-        if (temp_ret < phi_f2) {
-            *intersect = D_8016A740;
-            phi_f2 = temp_ret;
+    if (Math3D_CylVsLineSeg(cyl, &tri->vtx[0], &tri->vtx[2], &cylIntersectA, &cylIntersectB)) {
+        distFromPointAToIntersectASq = Math3D_Vec3fDistSq(&cylIntersectA, &tri->vtx[0]);
+        if (distFromPointAToIntersectASq < minDistSq) {
+            *intersect = cylIntersectA;
+            minDistSq = distFromPointAToIntersectASq;
         }
     }
     // what is 1.e38f ?
-    if (phi_f2 != 1.e38f) {
-        return 1;
+    if (minDistSq != 1.e38f) {
```

Comment:
1.e38f is probably "supposed" to be infinity since it's basically max float. And I don't think it needs a define or any comment to explain it.